File: src/main/java/it/contrader/dto/OrganizeSentenceDTO.java
package it.contrader.dto;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/*
**created by Torquato Di Maio
*/

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class OrganizeSentenceDTO {
	
	private static final String typeGame = "OrganizeSentence";
		
	private Long id;
	
	private String solution; 

	private String sentence; //frase disordinata
	
	private String definition;
	
	private CategoryDTO category;
	
	private LevelDTO level;

	public static String getTypeGame() {
		return typeGame;
	}
}

File: src/dto/organizesentencedto.ts

import {CategoryDTO} from './categorydto';
import {LevelDTO} from './leveldto';

/*
* created by Torquato Di Maio
*/

export class OrganizeSentenceDTO{
    static typeGame: string = "OrganizeSentence";

    id: number; //tipo numerico
	
    solution: string; //tipo stringa 
    
    sentence: string; //frase disordinata
	
    definition: string;
	
	category: CategoryDTO;
	
    level: LevelDTO;
    
    categoryTitle: string;
    levelName: string;
    
}